src/tests/unit/components/Envelope.vue.spec.js

```diff
@@ -0,0 +1,321 @@
+			mocks: {
+				$route,
+			},
+			propsData: {
+				account: {},
+				mailbox: {
+					specialRole:'',
+				},
+				data: {
+					myAcls: 's',
+					accountId: 123,
+					from: [{email:'info@test.com'}],
+					flags: { seen:false, flagged:false, $junk:false, answered:false, hasAttachments:false, draft:false, },
+				},
+			},
+			store,
+			localVue,
+		})
+
+		expect(view.vm.hasSeenAcl).toBe(true)
+	})
+	it('allows toggling archive action without ACLs', () => {
+		const view = shallowMount(Envelope, {
+			mocks: {
+				$route,
+			},
+			propsData: {
+				account: {},
+				mailbox: {
+					specialRole:'',
+				},
+				data: {
+					myAcls: undefined,
+					accountId: 123,
+					from: [{email:'info@test.com'}],
+					flags: { seen:false, flagged:false, $junk:false, answered:false, hasAttachments:false, draft:false, },
+				},
+			},
+			store,
+			localVue,
+		})
+
+		expect(view.vm.hasArchiveAcl).toBe(true)
+	})
+	it('source mailbox has t ACLs for archiving', () => {
+		const view = shallowMount(Envelope, {
+			mocks: {
+				$route,
+			},
+			propsData: {
+				account: {},
+				mailbox: {
+					specialRole:'',
+				},
+				data: {
+					myAcls: 't',
+					accountId: 123,
+					from: [{email:'info@test.com'}],
+					flags: { seen:false, flagged:false, $junk:false, answered:false, hasAttachments:false, draft:false, },
+				},
+			},
+			store,
+			localVue,
+		})
+
+		expect(view.vm.hasArchiveAcl).toBe(true)
+	})
+
+	it('source mailbox has e ACLs for archiving', () => {
+		const view = shallowMount(Envelope, {
+			mocks: {
+				$route,
+			},
+			propsData: {
+				account: {},
+				mailbox: {
+					specialRole:'',
+				},
+				data: {
+					myAcls: 'e',
+					accountId: 123,
+					from: [{email:'info@test.com'}],
+					flags: { seen:false, flagged:false, $junk:false, answered:false, hasAttachments:false, draft:false, },
+				},
+			},
+			store,
+			localVue,
+		})
+
+		expect(view.vm.hasArchiveAcl).toBe(true)
+	})
+	it('disallows toggling archive action without i ACL right', () => {
+		const view = shallowMount(Envelope, {
+			mocks: {
+				$route,
+			},
+			propsData: {
+				account: {},
+				mailbox: {
+					specialRole:'',
+				},
+				data: {
+					myAcls: 'x',
+					accountId: 123,
+					from: [{email:'info@test.com'}],
+					flags: { seen:false, flagged:false, $junk:false, answered:false, hasAttachments:false, draft:false, },
+				},
+			},
+			store,
+			localVue,
+		})
+
+		expect(view.vm.hasArchiveAcl).toBe(false)
+	})
+	it('allows toggling archive action with i ACL right', () => {
+		const view = shallowMount(Envelope, {
+			mocks: {
+				$route,
+			},
+			propsData: {
+				account: {},
+				mailbox: {
+					specialRole:'',
+				},
+				data: {
+					myAcls: 'i',
+					accountId: 123,
+					from: [{email:'info@test.com'}],
+					flags: { seen:false, flagged:false, $junk:false, answered:false, hasAttachments:false, draft:false, },
+				},
+			},
+			store,
+			localVue,
+		})
+
+		expect(view.vm.hasArchiveAcl).toBe(true)
+	})
+
```

Comment:
```suggestion
```

Test is redundant after changes above have been implemented. One positive test is enough.